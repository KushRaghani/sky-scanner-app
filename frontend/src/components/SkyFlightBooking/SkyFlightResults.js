import React from "react";
import PropTypes from "prop-types";
import SkyCalendar from "../SkyCalendar";

export const formatPricesData = data => {
  const a = {};
  for (var i in data) {
    var j = new Date(i).toUTCString();
    a[j] = data[i];
  }
  return a;
};
//Wrapper component for SkyCalendar
const SkyFlightResults = ({ data }) => {
  return (
    <React.Fragment>
      <SkyCalendar data={formatPricesData(data)} />
    </React.Fragment>
  );
};

SkyFlightResults.propTypes = {
  data: PropTypes.object.isRequired
};

export default SkyFlightResults;
