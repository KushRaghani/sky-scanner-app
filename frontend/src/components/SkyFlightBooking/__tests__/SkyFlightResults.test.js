import React from "react";
import { shallow } from "enzyme";
import SkyFlightResults from "../SkyFlightResults";

function loadSkyFlightResults(args) {
  const defaultProps = {
    data: {}
  };
  const props = { ...defaultProps, ...args };
  return shallow(<SkyFlightResults {...props} />);
}
describe("SkyFlightResults", () => {
  // Enyzme tests
  test("renders SkyFlightResults", () => {
      const wrapper = loadSkyFlightResults();
      expect(wrapper.find("SkyCalendar").length).toBe(1);
  });
});
