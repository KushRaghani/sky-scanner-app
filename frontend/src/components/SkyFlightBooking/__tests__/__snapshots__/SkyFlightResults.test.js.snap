// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkyFlightResults snapshot with no data 1`] = `
<div
  className="calendar"
>
  <div
    className="header calendar-row flex-middle"
  >
    <div
      className="block block-start"
    >
      <div
        className="icon"
        onClick={[Function]}
      >
        chevron_left
      </div>
    </div>
    <div
      className="block block-center"
    >
      <span>
        September 2019
      </span>
    </div>
    <div
      className="block block-end"
      onClick={[Function]}
    >
      <div
        className="icon"
      >
        chevron_right
      </div>
    </div>
  </div>
  <div
    className="calendar-days calendar-row"
  >
    <div
      className="block block-center"
    >
      Sunday
    </div>
    <div
      className="block block-center"
    >
      Monday
    </div>
    <div
      className="block block-center"
    >
      Tuesday
    </div>
    <div
      className="block block-center"
    >
      Wednesday
    </div>
    <div
      className="block block-center"
    >
      Thursday
    </div>
    <div
      className="block block-center"
    >
      Friday
    </div>
    <div
      className="block block-center"
    >
      Saturday
    </div>
  </div>
  <div
    className="main"
  >
    <div
      className="calendar-row"
    >
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          1
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          2
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          3
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          4
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          5
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          6
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          7
        </span>
      </div>
    </div>
    <div
      className="calendar-row"
    >
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          8
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          9
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          10
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          11
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          12
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          13
        </span>
      </div>
      <div
        className="block cell block-val selected"
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          14
        </span>
      </div>
    </div>
    <div
      className="calendar-row"
    >
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          15
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          16
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          17
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          18
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          19
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          20
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          21
        </span>
      </div>
    </div>
    <div
      className="calendar-row"
    >
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          22
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          23
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          24
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          25
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          26
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          27
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          28
        </span>
      </div>
    </div>
    <div
      className="calendar-row"
    >
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          29
        </span>
      </div>
      <div
        className="block cell block-val "
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          30
        </span>
      </div>
      <div
        className="block cell block-val disabled"
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          1
        </span>
      </div>
      <div
        className="block cell block-val disabled"
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          2
        </span>
      </div>
      <div
        className="block cell block-val disabled"
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          3
        </span>
      </div>
      <div
        className="block cell block-val disabled"
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          4
        </span>
      </div>
      <div
        className="block cell block-val disabled"
        onClick={[Function]}
      >
        <span
          className="calendar-date"
        >
          5
        </span>
      </div>
    </div>
  </div>
</div>
`;
